core/src/main/scala/org/apache/spark/internal/config/UI.scala

```diff
@@ -92,6 +92,11 @@ private[spark] object UI {
     .booleanConf
     .createWithDefault(true)
 
+  val UI_HEAP_DUMPS_ENABLED = ConfigBuilder("spark.ui.threadDumpsEnabled")
+    .version("3.4.0")
+    .booleanConf
+    .createWithDefault(true)
```

Comment:
IMO the default should be `false`.
Heap dump causes "stop the world" in JVM and it might lead to security risks if it is enabled by default.

Reply:
Modified